src/curl.c

```diff
@@ -311,6 +311,99 @@ CURLcode swupd_download_file_complete(CURLcode curl_ret, struct curl_file *file)
 	return curl_ret;
 }
 
+
+// err is only needed because retry code is spread all over swupd. Remove this error code return when
+// #211 is fixed
+enum download_status process_curl_error_codes(int curl_ret, CURL *curl_handle, int *err)
+{
+	char *url;
+	if (curl_easy_getinfo(curl_handle, CURLINFO_EFFECTIVE_URL, &url) != CURLE_OK) {
+		url = "<not available>";
+	}
+
+	if (curl_ret == CURLE_OK || curl_ret == CURLE_HTTP_RETURNED_ERROR) {
+		long response = 0;
+		if (curl_ret == CURLE_OK || curl_ret == CURLE_HTTP_RETURNED_ERROR) {
```

Comment:
This condition is identical to the one on line 324, so it will always evaluate to true.

Reply:
True. this is a left over. I'll remove this conditional